### DISCUSSION TITLE
Deploying a production Single Page Application?

### DISCUSSION THREAD

**Discussion by sampalm** [wailsapp/wails#2132] [Q&A]
I know that the `wails dev` host a web version for development purpose only, but, is there a way to have that web version hosted on a production environment? 

I must be missing something here, but from what I've seen if I build my application around Wails, and want to deploy it to the cloud, I'll lose the ability to talk with my Wails backend, therefore I will have to change all those calls, there's a way to build a Single Page Application based on my Wails application similar to what we already have in Ionic?

**Comment by leaanthony** [comment]
Wails wasn't initially designed for this use case but it is something more and more people are asking for so we are looking into it. To be honest, it's not my area of expertise so the more we have in that discussion the better. Is that something you're willing to contribute to?

**Reply by sampalm** [reply]
Sure, I think Wails have huge potential, if I can I'd like to contribute to it.

**Reply by leaanthony** [reply]
Brilliant! I'm just trying to organise a bit more around enabling people to better contribute, better process etc. For now, feel free to contribute thoughts to this ticket: https://github.com/wailsapp/wails/issues/1940

**Comment by fairking** [comment]
You could have a logic based on the environment variables:
```js
window.make_request("store/create_item", json_data) {
    if (env.spa) {
        // call http
    } else {
        // call wails
    }
}
```

Please note, this is just some not working example. You must find that way by yourself.

app.go
```go
// Request model
type RequestModel struct {
	Route string
	Body string
}

// Response model
type ResponseModel struct {
	Response string
	Error string // Validation error
}

// Request from the frontend
func (a *App) Request(data RequestModel) (*ResponseModel, error /* System error */) {
	return &ResponseModel{ Response: "ABC" }, nil
}
```
